.azure-pipelines/templates/steps/tox-steps.yml

```diff
@@ -41,6 +41,9 @@ steps:
       ln -s $(testFarmPem.secureFilePath) tests/letstest/test-farm.pem
     condition: contains(variables['TOXENV'], 'test-farm')
   - bash: |
+      export TARGET_BRANCH="${BUILD_SOURCEBRANCHNAME}"
```

Comment:
I don't think this works reliably for branch names containing forward slashes. For example, see [this](https://dev.azure.com/certbot/certbot/_build/results?buildId=2224&view=logs&j=723b2a1e-037a-5f62-87ff-97151358ff8d&t=783bcbc0-3b86-5dca-1277-043d2796568b&l=135) where the name of the branch was `test-/full-azure-pipelines`.

That branch name is clearly pathological, but it's not unheard of to have branch names containing a slash this. There are a few on the Certbot repo right now. If we set `TARGET_BRANCH` to the wrong value, we'll try to fetch and compare things from a different branch which I think can easily result in a confusing error which I think we should avoid in some way here.

I'm not sure if `SYSTEM_PULLREQUEST_TARGETBRANCH` has the same problem. If so, maybe we should fix that too, however, I think it is much less likely to be a problem.

Maybe we could try to use `Build.SourceBranch` and strip off leading `/refs/heads/` or `/refs/tags/`? Maybe we should reconsider just setting this for tags? What do you think?

Reply:
I used `Build.SourceBranch` with inplace variable substitution. Since it does not add a significant complexity I think we can continue to try reaching the behavior in Travis. I expect that `SYSTEM_PULLREQUEST_TARGETBRANCH` will have the same flaw, but I agree that it is very unlikely that it would happen in this case.

Reply:
This doesn't seem to work for some reason. See

* https://dev.azure.com/certbot/certbot/_build/results?buildId=2231&view=logs&j=d604148c-357f-54f5-f721-8901e97c14e9&t=ecd66334-46bc-51f3-5bf9-92dfed1b03e8&l=110
* https://dev.azure.com/bmw0523/bmw/_build/results?buildId=259&view=logs&j=723b2a1e-037a-5f62-87ff-97151358ff8d&t=91ba8e50-6288-595b-d759-47123c13bab4&l=135.

Reply:
Ok, now it is: https://dev.azure.com/certbot/certbot/_build/results?buildId=2235&view=logs&j=8d8187c4-68b0-590f-5457-0716907b8b55&t=c70ceb87-faab-5954-8e61-9a1f3f9a26a0&l=110